vulcanai/models/metrics.py

```diff
@@ -741,8 +741,174 @@ def _run_test_multi(network, data_loader, plot=False, save_path=None,
+            valid_interv : int
+                Specifies after how many epochs validation should occur.
+            plot : boolean
+                Whether or not to plot all results in prompt and charts.
+	    index_to_iter : string
+                Index of feature within data_loader who's values will be iterated to assess difference
+            ls_feat_vals : list
+                List of values for feature provided in index_to_iter
+
+	Returns: 
+	    improvement_score : float
+        """
+        dct_improvementScores = defaultdict(list)
+        dct_filteredSubj = defaultdict(list)
+        ls_filteredProbs = []
+	
+        fold_len = math.floor(len(data_loader.dataset) / k)
+        rem = len(data_loader.dataset) % k
+        fold_seq = []
+
+        for _ in range(k-1):
+            fold_seq.append(fold_len)
+        if rem == 0:
+            fold_seq.append(fold_len)
+        else:
+            fold_seq.append(fold_len+rem)  # last one is the longest if unequal
+
+        dataset_splits = data.random_split(data_loader.dataset,
+                                           fold_seq)
+
+        batch_size = data_loader.batch_size
+        shuffle = isinstance(data_loader.sampler, data.sampler.RandomSampler)
+
+        try:
+            for fold in range(k):
 
-    # TODO: include support
+                # TODO: this may break on different devices?? test.
+                # TODO: Re-initialize instead of deepcopy?
+                cross_val_network = copy.deepcopy(network)
+
+                # TODO: properly pass params
+
+                # Generate fold training set.
+                train_dataset = data.ConcatDataset(
+                    dataset_splits[:fold] + dataset_splits[fold+1:])
+                # Generate fold validation set.
+                val_dataset = dataset_splits[fold]
+                # Generate fold training data loader object.
+                train_loader = data.DataLoader(
+                    train_dataset, batch_size=batch_size, shuffle=shuffle)
+                # Generate fold validation data loader object.
+                val_loader = data.DataLoader(
+                    val_dataset, batch_size=batch_size)
+                # Train network on fold training data loader.
+                cross_val_network.fit(
+                    train_loader, val_loader, epochs,
+                    retain_graph=retain_graph,
+                    valid_interv=valid_interv, plot=plot, save_path=save_path)
+                dct_scores = get_probs(network, val_loader, index_to_iter, ls_feat_vals)
+                dct_filtered = filter_matched_subj(dct_scores, val_loader, index_to_iter)
+                for val in list(dct_filtered):
+                    dct_filteredSubj[val].append(dct_filtered[val])
+                    ls_filteredProbs.append(dct_filtered[val])
+            V_p = np.array(ls_filteredProbs).mean()
+            ls_pos_rate = [subj for subj in data_loader.dataset if subj[1].item() == 1]
```

Comment:
what is this doing?

Reply:
`ls_pos_rate` essentially just creates a list of subjects (probably should rename) who belong to the positive label in the data_loader. It is then used for getting the percentage of positive label subjects to calculate the improvement score.